### ISSUE TITLE
some questions about visual feature extractor 

### ISSUE THREAD

**Issue by YujieOuO** [skelemoa/synse-zsl#11]
Hello, I follow this paper and use shift-gcn to extract the visual features.
There are some questions.
1、I use the seen classes to train the shift-gcn, but without the test data, how can i choose the best epoch?
2、For each stream in shift-gcn, i can get a type of visual embedding. So how to get the 4s embeddding? just add them up in a ratio of [0.6,0.6,0.4,0.4]?
3、the best.pt in synse_resources/ntu_results/shift_5_r/weights/ is the weight for all the 4s shift-gcn models?
(maybe each stream has a best.pt?)

thank you for your work

**Comment by pranaygupta36** [comment]
1. For training the seen classifier, we started with the cross subject(CS) splits of the datasets, and used the seen class samples from the train CS split and the tested using the samples of the seen classes from the test CS split.
2-3. Since, the other streams only bring a small increment in the classification performance, we only use the model trained on joint stream. The best.pt provided is corresponding to that only.

**Comment by YujieOuO** [comment]
oh,i see. Thank you for your reply. @pranaygupta36 @cvitatma @AnyiRao
So, i follow the Data Preparation in Shift-GCN. 
Then in xsub folder(CS split), train_data_joint.npy and val_data_joint.npy are merged together to acquire all joint data. 
After that, unseen classes ([10,11,19,26,56])  are picked out and the rest are seen classes.
Next, using shift-gcn.py with pre-trained best.pt, i could acquire the visual features for seen and unseen classes.Is it correct?
In split5,the number of all data is 56578. and i find the number of seen classes is 51877 and unseen classses is 4701. 
But in your floder(synse_resources/ntu_results/shift_5_r/) , the number of seen classes in train.npy is 36548 and unseen classes in ztest.npy is 1356. Did you do other data preparation?

**Comment by pranaygupta36** [comment]
Yes, the splits are actually created from the cross subject (cs) split suggested by the NTU datasets, where training samples from the seen classes are selected from the train cs split and zsl test samples are selected from the test (cs) split. The test samples of the seen classes are used for testing while training the visual classifier (shift gcn).

**Comment by YujieOuO** [comment]
so, for zsl, the seen classes are from the train data in CS and the unseen classes are from the test data in CS? Why don't you select the seen and unseen classes from all joint data? you know in split5, the number of unseen classes is 4701 from all joint data, but you just use 1356 unseen classes.

**Comment by pranaygupta36** [comment]
The test CS split data from the seen classes is used for testing the visual classifier (Shift-GCN) training. But you make valid point about the unseen class training data. It is not utilized anywhere and should have been used for zsl.

**Comment by ruczhouyujie** [comment]
> The test CS split data from the seen classes is used for testing the visual classifier (Shift-GCN) training. But you make valid point about the unseen class training data. It is not utilized anywhere and should have been used for zsl.

yes, actually it is strange the number of unseen class is only 1356. And the true number of unseen classes in NTU-60 could be 4701. The amount of data in the zsl test phase will seriously affect accuracy. Will you change it and do the experiment again?
